Add test for issue 4845

marberi opened the pull request:
- Tests added for issue 4845

First time adding a test. I could run pytest, but it did not find this new test. Check
carefully.

GPUtester commented:
Can one of the admins verify this patch?

Admins can comment `ok to test` to allow this one PR to run or `add to allowlist` to allow all future PRs from the same author to run.

marberi commented:
@jakirkham

phobson commented:
@marberi thanks for the PR and the test.

Based on my understanding of the underlying issue, I'd expect this to fail. Indeed, when I run it locally in isolation (not a part of the larger suite), it fails by raising and error related to mult-indexes.

Is your intent here to show/confirm/document that multi-indexes are not supported? If so, we should decorate the tests with an xfail mark. Something like:

```python
@pytest.mark.xfail(reason="Multi-indexes are not supported")
```

Does that make sense? Is this your intention with this PR?

marberi commented:
@phobson The original goal was simply adding a test, since I thought it was fixed and @jakirkham seemed to confim that in https://github.com/dask/dask/issues/4845 . Now, running the test manually (pytest test_base.py), I was able to trigger the error. However, the error did not occur when running as a script, including setting scheduler='multiprocessing'. Do you understand what can be the difference?

Adding a failing test and marking them as a known issue seems like a good solution. After submitting this patch, I realized that there is another test directory with dataframe related test. If resubmitting this, do I add another commit on the same branch or forcefully push a change ignoring the old commit? Thanks for the feedback. I am unsure how you normally handle this in larger projects.

phobson commented:
@marberi no need to force-push. Add as many PR-related commits as your need to the existing and push to the same branch. Github will recognize the new commits and update the PR accordingly. When PRs in the dask project are merged, all of the commits are "squashed" into a single commit.

phobson commented:
To your point about other test directories, I think dask/dataframe/tests/test_groupby.py is the most logical home for this test

ian-r-rose commented:
add to allowlist

ian-r-rose commented on dask/tests/test_base.py:
```diff
@@ -983,6 +983,17 @@ def test_persist_scalar_rename():
+    df = pd.DataFrame({"ind_a": np.arange(100), "ind_b": 1, "var": "whatever"})
+    df = dd.from_pandas(df, npartitions=90)
+
+    # Only fails when grouping with two variables.
```
I think this comment is likely to be confusing to people who haven't read the initial issue. Maybe something like "in older versions of pandas, this would fail due to a thread-safety issue"

ian-r-rose commented on dask/tests/test_base.py:
```diff
@@ -983,6 +983,17 @@ def test_persist_scalar_rename():
     dd.utils.assert_eq(dds2, 5)
 
 
+@pytest.mark.skipif("not dd or not np")
```
If you put this with the rest of the dataframe groupby tests, then this skip shouldn't be necessary.

ian-r-rose reviewed (commented):
Thanks for adding this @marberi! I agree with @phobson that `test_groupby.py` would be a more natural place for this test.

marberi commented:
Moved this over to test_groupy.py and removed the "np+pd" dependencies. When originally submitting I was not aware of this subdirectory. Also, when originally submitting this request, the problem did not occur when simply running the test code. Following @phobson suggestion, I add a "pytest.mark.xfail" to mark that this fails. 

(tests are still running)
